From 8a5d7ec9457eee6dbf74a74c9064b8f9eaf90461 Mon Sep 17 00:00:00 2001
From: Mike Looijmans <mike.looijmans@topic.nl>
Date: Tue, 8 Dec 2020 08:16:27 +0100
Subject: [PATCH] zynqmp-topic-miamiplusmp.dts: Proper PHY reset timing

Whereas most ethernet PHYs require pulses of a few microseconds, the Marvell
PHY requires a 10ms reset pulse and a wait time of 50ms after that. Specify
this in the devicetree as the defaults won't do for this chip.
---
 arch/arm64/boot/dts/xilinx/zynqmp-topic-miamiplusmp.dts | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/arch/arm64/boot/dts/xilinx/zynqmp-topic-miamiplusmp.dts b/arch/arm64/boot/dts/xilinx/zynqmp-topic-miamiplusmp.dts
index cdf62dab41d0..f09150e2fbc1 100644
--- a/arch/arm64/boot/dts/xilinx/zynqmp-topic-miamiplusmp.dts
+++ b/arch/arm64/boot/dts/xilinx/zynqmp-topic-miamiplusmp.dts
@@ -394,6 +394,9 @@
 			device_type = "ethernet-phy";
 			reg = <0x1>;
 			reset-gpios = <&gpioex 2 1>; /* Active low, on port 2 of GPIO expander */
+			/* Datasheet: 10 ms reset pulse, wait 50 ms for register access */
+			reset-assert-us = <10000>;
+			reset-deassert-us = <50000>;
 			interrupt-parent = <&gpio>;
 			interrupts = <27 IRQ_TYPE_LEVEL_LOW>;
 			marvell,reg-init =
-- 
2.17.1

